lib/read_preference.js

```diff
@@ -5,7 +5,7 @@
  * used to construct connections.
  * @class
  * @param {string} mode A string describing the read preference mode (primary|primaryPreferred|secondary|secondaryPreferred|nearest)
- * @param {array} tags The tags object
+ * @param {array} [tags] The tags object
```

Comment:
While we're here, maybe we can improve these doc strings? "The tags object" isn't really helpful, and it documented as accepting an `Array`

Reply:
I just happened to be editing this file for ts stuff so for ease of life sake:
`A tag set used to target reads to members with the specified tag(s). tagSet is not available if using read preference mode primary.`
probably `{object[]}` is the best type

Reply:
We have tests which likely need to change because of this btw, e.g. `replset_read_preference.test.js` does:
```
new ReadPreference(ReadPreference.SECONDARY, { loc: 'ny' })
```

Reply:
Here's another consideration. `read_preference.test.js` does:
```
new ReadPreference('nearest', { maxStalenessSeconds: 100 })
```
Which would imply that our method overloads for construction of this type are:
```
constructor(selector, tags, options);
constructor(selector, options);
constructor(selector);
```
I think we should have tests around all possible variations

Reply:
@mbroadst Just checking, are you saying we want to continue supporting the `constructor(selector, options)` overload? I thought that part of this effort was to deprecate that construction style going forward, but I can certainly put support for it back in.

Reply:
> We have tests which likely need to change because of this btw, e.g. `replset_read_preference.test.js` does:
> 
> ```
> new ReadPreference(ReadPreference.SECONDARY, { loc: 'ny' })
> ```

All of the tests that would fail are actually currently disabled, as far as I can see. The full `npm run test` currently passes with this implementation.

Reply:
@emadum we do want to support that form of construction, sorry if that wasn't clear in the (minimal) language of the ticket! The intent of the validation was to identify and correct instantiations like:
```
new ReadPreference(ReadPreference.SECONDARY, { loc: 'ny' })
```

Given that y'all have just enabled Mocha testing in VSCode, this might be a really good opportunity to clean up `./functional/readpreference.test.js`, ensuring it shows up correctly in the column (using `context` blocks for instance)  and that we have a full coverage over the type. How does that sound?

Reply:
Sounds good, thanks for the clarification; agreed it's a good time to clean up the tests.